### ISSUE TITLE
No way to check if prefix/suffix are passed

### ISSUE THREAD

**Issue by rvmourik** [wearebraid/vue-formulate#489]
**Describe the bug**
Passing a suffix/prefix works in general, but there is no way to check for the presents in a class styling method from the config. Is it simply not present in the classContext variable that is passed to those callback functions.

**Expected behavior**
A way to check if a suffix or prefix is present on an element so additional classes can be applied to the element. So for example you can add some padding to the input so a icon that was prepended is aligned next to the text.

I did not found an option to do so, also by looking through the codebase.

Thanks, if more info is needed please let me know.

**Comment by justin-schroeder** [comment]
By "passed" do you mean in the if the scoped slot is used or not?

**Comment by rvmourik** [comment]
> By "passed" do you mean in the if the scoped slot is used or not?

Yes, I worked around it with this:

```vue
<script>
  export default {
    name: 'WsFormInputAddonSuffix',
    extends: Base,
    mounted() {
      const $parent = this.$parent;

      if ($parent.$el && $parent.$el.querySelector('input')) {
        const input = this.$parent.$el.querySelector('input');

        input.classList.add('form__input--addition-suffix');
        input.classList.add(`form__input--addition-suffix-${this.type}`);
      }
    }
  };
</script>
```

I created two components for the Suffix and Prefix slot, from within this custom component I add classes to the input that is within the parent. It works, but it not as clean as needed I guess.

So if a cleaner way could be created in the library that would be awesome :-)

**Comment by rvmourik** [comment]
@justin-schroeder I saw another "weird" thing working with the suffix/prefix slots, I added a suffix slot to a box-element. Looking in the source I expected the suffix to be rendered after the input (or label). Instead it renders the suffix in the group that is the parent of the two box elements.

**Code**
```js
          <formulate-input
            type="radio"
            name="shipToType"
            element-class="flex items-center space-x-3 mt-1"
            :wrapper-class="['font-medium']"
            :options="shipToOptions"
            ignored
          >
            <template #suffix>
              <div class="font-base text-sm text-gray-500">suffix is here</div>
            </template>
          </formulate-input>
```

**How it looks in the source**
![image](https://user-images.githubusercontent.com/6427451/134668649-814ebe1c-9ab7-4242-a529-f30b9b9cc1dd.png)

Did I do something wrong, or is this expected or is it a bug?

**Comment by rvmourik** [comment]
@justin-schroeder any chance you saw the last message regarding the box element? Would love some input if that's possible, thanks in advance Justin.

**Comment by justin-schroeder** [comment]
yeah, unfortunately I think we would need to expose a new type of slot for that — the suffix is the suffix of the group of inputs, but it makes sense that sometimes you want to define it as the suffix of the actual input. We have much better ways to do this coming out in [FormKit](https://formkit.com) — but we could certainly add something like `boxPrefix` or `boxSuffix` to Vue Formulate as a stopgap.

**Comment by rvmourik** [comment]
> yeah, unfortunately I think we would need to expose a new type of slot for that — the suffix is the suffix of the group of inputs, but it makes sense that sometimes you want to define it as the suffix of the actual input. We have much better ways to do this coming out in [FormKit](https://formkit.com) — but we could certainly add something like `boxPrefix` or `boxSuffix` to Vue Formulate as a stopgap.

If such a solution is something that is possible for the time being that would be great, and good to hear such improvements are part of the core in FormKit. 

Is there any date for a new release for Vue Formulate?
